app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -475,18 +476,18 @@ protected function _getOrdersTrackingCodeAnalytics4()
                     $_item = [
                         'item_id' => $item->getSku(),
                         'item_name' => $item->getName(),
-                        'quantity' => $item->getQtyOrdered(),
-                        'price' => $item->getBasePrice(),
-                        'discount' => $item->getBaseDiscountAmount()
+                        'quantity' => intval($item->getQtyOrdered()),
+                        'price' => number_format($item->getBasePrice(), 2),
```

Comment:
Does GA read prices correctly if they includes commas?

Reply:
I think only dot is supported

Reply:
`number_format` will add in a comma to separate thousands

Reply:
number_format is tested

Reply:
I think round removed the decimals if it rounds to zero, while it's better for GA to keep the fixed number of zeros, which only number_formats provides :-)

Reply:
Honestly, I don't know, but i' have no time to try with an amount like 1999.99 to see if it generates any errors in GA4 reports.